add k8s docs for getting started, K8s Manifest and Helm (#179)
opea-project/docs: examples/ChatQnA/deploy/k8s_manifest.md

# Multi-node on-prem deployment with TGI on Xeon Scalable processors on a K8s cluster using Helm Charts
<pod-name> in helm update

> This deployment section covers multi-node on-prem deployment of the ChatQnA example with OPEA comps to deploy using the TGI service. There are several slice-n-dice ways to enable RAG with vectordb and LLM models, but here we will be covering one option of doing it for convenience: we will be showcasing how to build an e2e chatQnA with Redis VectorDB and neural-chat-7b-v3-3 model, deployed on a Kubernetes cluster using Helm. For more information on how to setup a Xeon based Kubernetes cluster along with the development pre-requisites, follow the instructions here [Kubernetes Cluster and Development Environment](./k8s_getting_started.md#kubernetes-cluster-and-development-environment).

Reviewer: Quoting or highlighting names where relevant, and/or splitting this to multiple paragraphs, could make this long text more readable.

> For a quick introduction on deploying using Mainfest files, visit the  `Using Kubernetes Manifest to Deploy`  section in  [Getting Started with Kubernetes for ChatQnA](./k8s_getting_started.md)

Reviewer: Do manifests need a mention? (they're likely to be dropped for 1.2)

Author: @eero-t removed all references to manifests

## Overview

There are several ways to setup a ChatQnA use case. Here in this tutorial, we
will walk through how to enable the below list of microservices from OPEA
[GenAIComps](https://github.com/opea-project/GenAIComps) to deploy a multi-node TGI megaservice solution.

1. Data Prep
2. Embedding
3. Retriever
4. Reranking
5. LLM with TGI

> > **Note:** ChatQnA can also be deployed on a single node using Kubernetes, provided that all pods are configured to run on the same node.

Reviewer: And it has resources (memory) for running all of them...

Author: @eero-t included this point

## Prerequisites

### Clone Repository 
To set up the workspace for deploying ChatQnA via Kubernetes, start by cloning the `GenAIExamples` repository and navigate to the ChatQnA Kubernetes manifests directory:

```bash
https://github.com/opea-project/GenAIExamples.git
```
Checkout the release tag
```
cd GenAIExamples/ChatQnA/kubernetes/intel/cpu/xeon/manifest
git checkout tags/v1.1
```
### Bfloat16 Inference Optimization

> We recommend using newer CPUs, such as 4th Gen Intel Xeon Scalable processors (code-named Sapphire Rapids) and later, that support the bfloat16 data type. If your hardware includes such CPUs and your model is compatible with bfloat16, adding the `--dtype bfloat16` argument to the HuggingFace `text-generation-inference` server can significantly reduce memory usage by half and provide a moderate speed boost. This change has already been configured in the `chatqna_bf16.yaml` file. To use it, follow these steps:

Reviewer: ```suggestion
We recommend using newer CPUs, such as 4th Gen Intel Xeon Scalable processors (code-named Sapphire Rapids) and later, that support the bfloat16 data type. If your hardware includes such CPUs and your model is compatible with bfloat16, adding the `--dtype bfloat16` argument to the HuggingFace `text-generation-inference` server halves its memory usage and provides a moderate speed boost. This change has already been configured in the `chatqna_bf16.yaml` file. To use it, follow these steps:
```

Author: @eero-t will remove manifests related files given that it might be dropped

Run `kubectl get nodes` and identify the nodes in your cluster with BFloat16 support and label the nodes to schedule the service on it automatically:

```bash
kubectl label node <node-name> node-type=node-bfloat16
```

> >**Note:**  The manifest folder has several configuration pipelines that can be deployed for ChatQnA. In this example, we'll use the `chatqna_bf16.yaml` configuration. You can use `chatqna.yaml` instead if you don't have BFloat16 support in your nodes.

Reviewer: With Helm charts, one can e.g. enable the relevant BF16 option in the values file: https://github.com/opea-project/GenAIInfra/blob/main/helm-charts/common/tgi/values.yaml#L23

Author: @eero-t Added helm bf16 instructions
